src/main/kotlin/com/autonomousapps/internal/utils/gradleStrings.kt

```diff
@@ -50,30 +50,25 @@ private fun ProjectComponentIdentifier.identityPath(): String {
 }
 
 internal fun ResolvedArtifactResult.toCoordinates(): Coordinates {
-  val gradleVariantIdentification = variant.toGradleVariantIdentification()
-  val resolved = id.componentIdentifier.toCoordinates(gradleVariantIdentification)
+  return variant.wrapInIncludedBuildCoordinates(id.componentIdentifier)
+}
+
+private fun ResolvedVariantResult.wrapInIncludedBuildCoordinates(id: ComponentIdentifier): Coordinates {
+  val variantIdentification = toGradleVariantIdentification()
+  val resolved = id.toCoordinates(variantIdentification)
 
   // Doesn't resolve to a project, so can't be an included build. Return as-is.
   if (resolved !is ProjectCoordinates) return resolved
 
-  // may be a composite substitution
-  val identity = ProjectCoordinates(
-    (id.componentIdentifier as ProjectComponentIdentifier).identityPath(),
-    gradleVariantIdentification
-  )
-
-  // Identity path matches project path, so we assume this isn't resolved from an included build, and return as-is.
-  if (resolved == identity) return resolved
-
-  // At this point, we think this module has resolved from an included build.
-
+  // Module may have been resolved from an included build. Construct IncludedBuildCoordinates if possible.
   // This is a very naive heuristic. Doesn't work for Gradle < 7.2, where capabilities is empty.
-  val requested = variant.capabilities.firstOrNull()?.let { c ->
+  val projectName = (owner as ProjectComponentIdentifier).projectName
+  val requested = capabilities.find { it.name.startsWith(projectName) }?.let { c ->
     c.version?.let { v ->
       ModuleCoordinates(
-        identifier = "${c.group}:${c.name}",
+        identifier = "${c.group}:$projectName",
```

Comment:
In a sane setup, the "projectName" corresponds to the second element of the GA coordinate.

Using the capability here is not correct. We missed it before. For test fixtures for example, this is wrong because then `project-name-test-fixtures` would be used as identifier. We just did not hit this bug before because most tests were done from the "root" build. The existing tests captured this after the change.

Only the group is still guesswork then. But in normal project structures, it should be correct like this. I will open an issue at Gradle that the coordinates should be provided in the result here.

Reply:
Rediscovered: https://github.com/gradle/gradle/issues/20048 again. Will comment on that one.

Reply:
> In a sane setup, the "projectName" corresponds to the second element of the GA coordinate.

Can you elaborate on this? Because most Android projects I've seen are not "sane" in this way. They almost always have some random project name and custom artifact ID because (1) Android devs do not care about sane builds and (2) Gradle docs are confusing.

Reply:
Sorry for the sloppy wording.

If you change the artifact ID for publishing (which I believe is what you refer to) it has no influence here. The identity will be different, when you use the published version of the component. But as long as the component is a project in one of the builds, such a reconfiguration has no effect on what this plugin is looking at.

The only situation where the project name would not correspond to the "artifact name" is when the included build substitution is changed. This is a feature of Gradle I am not very happy with that it exists at all. :) But that problem already exists in the current plugin release. And adding more information to the resolution result for these cases is what https://github.com/gradle/gradle/issues/20048 is about.

Reply:
Ah, I meant in part publishing, but also the included build substitution thing. I've had to walk folks through that so many times. There's a good (bad) example of that in the build where I work. All the modules are named `:feature1:public`, `:feature2:public`, and so on. So if went with the default, then every projectName is just "public", which of course doesn't work.